micro-rdk/src/common/provisioning/server.rs

```diff
@@ -391,6 +469,112 @@ impl WifiManager for () {
     }
 }
 
+#[cfg(feature = "native")]
```

Comment:
not too happy with this but I wanted to avoid duplicating code for each variant and using generics became quickly too contrived and unreadable.

Reply:
the issue is that multiple bounds have to be added to make executor work with HyperExec trait  and provisioningExec traits so we would end up with a lot of bounds which make things harder to read. At this point we can merge NativeExecutor and Esp32Excutor so we should be able to make thing a bit easier to use. But that for some future work

Reply:
Sounds reasonable to me